src/components/ProjectSettingsTabs/GeneralTab/Fields/RepoConfigField.test.tsx

```diff
@@ -0,0 +1,255 @@
+  formData: {
+    owner: "evergreen-ci",
+    repo: "logkeeper",
+  },
+};
+
+const Field = () => (
+  <MockedProvider mocks={[attachProjectToRepoMock, detachProjectFromRepoMock]}>
+    <RepoConfigField
+      {...fieldProps}
+      uiSchema={{
+        options: {
+          projectId: "evergreen",
+          repoName: "evergreen",
+          repoOwner: "evergreen-ci",
+          useRepoSettings: true,
+        },
+      }}
+    />
+  </MockedProvider>
+);
+
+const AttachmentModal = ({
+  shouldAttach = true,
+}: {
+  shouldAttach?: boolean;
+}) => (
+  <MockedProvider mocks={[attachProjectToRepoMock, detachProjectFromRepoMock]}>
+    <AttachDetachModal
+      handleClose={() => {}}
+      open
+      projectId="evergreen"
+      repoName={fieldProps.formData.repo}
+      repoOwner={fieldProps.formData.owner}
+      shouldAttach={shouldAttach}
+    />
+  </MockedProvider>
+);
+
+describe("repoConfigField", () => {
+  it("does not show the move repo button when not attached to repo", async () => {
+    const { queryByDataCy } = render(
+      <RepoConfigField
+        {...fieldProps}
+        uiSchema={{
+          options: { useRepoSettings: false },
+        }}
+      />
+    );
+    expect(queryByDataCy("move-repo-button")).not.toBeInTheDocument();
+  });
+
+  it("clicking the button opens the modal", async () => {
+    const { queryByDataCy } = render(
+      <RepoConfigField
+        {...fieldProps}
+        uiSchema={{
+          options: { useRepoSettings: true },
+        }}
+      />
+    );
+    expect(queryByDataCy("move-repo-modal")).not.toBeInTheDocument();
+
+    const moveRepoButton = queryByDataCy("move-repo-button");
+    await fireEvent.click(moveRepoButton);
+    await waitFor(() => expect(queryByDataCy("move-repo-modal")).toBeVisible());
+  });
+
+  describe("moveRepoModal", () => {
+    it("renders the Move Repo Modal when the open prop is true", () => {
+      const mockOnConfirm = jest.fn();
+      const mockOnCancel = jest.fn();
+      const { queryByDataCy } = render(
+        <MoveRepoModal onCancel={mockOnCancel} onConfirm={mockOnConfirm} open />
+      );
+      expect(queryByDataCy("move-repo-modal")).toBeVisible();
+    });
+
+    it("does not render the Move Repo Modal when the open prop is false", () => {
+      const mockOnConfirm = jest.fn();
+      const mockOnCancel = jest.fn();
+      const { queryByDataCy } = render(
+        <MoveRepoModal
+          onCancel={mockOnCancel}
+          onConfirm={mockOnConfirm}
+          open={false}
+        />
+      );
+      expect(queryByDataCy("move-repo-modal")).not.toBeInTheDocument();
+    });
+
+    it("disables the confirm button on initial render", () => {
+      const mockOnConfirm = jest.fn();
+      const mockOnCancel = jest.fn();
+      const { getAllByText } = render(
+        <MoveRepoModal onCancel={mockOnCancel} onConfirm={mockOnConfirm} open />
+      );
+
+      const moveRepoButton = getAllByText("Move Repo")[1].closest("button");
```

Comment:
does `getByRole("button", { name: "Move Repo" })` work instead? https://testing-library.com/docs/queries/byrole/

Reply:
It helps target the button specifically without needed to rely on indexes since there are multiple elements named "Move Repo"